scripts/compare_warehouse_tables.py

```diff
@@ -0,0 +1,244 @@
+from datetime import datetime, timedelta
+from google.cloud import bigquery
+from tabulate import tabulate
+import pandas as pd
+import argparse
+import warnings
+
+# TODO Setup For Running The Code:
+# 1. Make sure to update warehouse_dev:
+#   A. Go to https://fcprod-jenkins.dsp-techops.broadinstitute.org
+#   B. Go to the Data Warehouse tab (if applicable)
+#   C. Click the WIP-data-warehouse-python-development job
+#   D. Check if code is pulling from */master branch (Info found in Configure Section, Under Source Code Management -> Branches to build)
+#   E. In the WIP-data-warehouse-python-development job, press Build Now
+# 2. Run "gcloud auth login" with firecloud.org email
+# 3. Run "gcloud auth application-default login" with firecloud.org email
+# 4. Run "gcloud auth application-default set-quota-project broad-dsde-dev"
+
+
+def call_bigquery(bq, env_query, project_name, verbose=False):
+    """Call BigQuery table."""
+    # Start job
+    query = bq.query(env_query)
+    if verbose:
+        print("Starting job.")
+
+    # Gather results
+    results = query.result()
+    if verbose:
+        print("Job finished.")
+
+    # read RowIterator object into pandas df
+    results_df = results.to_dataframe()
+    if verbose:
+        print(results_df)
+    return results_df
+
+
+def get_tables(schema, verbose=False):
+    """Get a list of the tables in warehouse_dev."""
+    # Getting the list of tables
+    dev_tables = list(schema['table_name'])
+
+    # Remove tables not in Prod or too long to parse
+    dev_tables.remove("rawls_entity")
+    dev_tables.remove("rawls_entity_attribute")
+    dev_tables.remove("cromwell_metadata")
+    if verbose:
+        print(f"List of dev tables in {dev_tables}")
+    return dev_tables
+
+
+def get_newrows(bq, project, table_names, tables_with_timestamps, verbose):
+    """Getting the table and information of new rows in dev and prod."""
+    # Creating the master list of the new rows in dev and prod for the table
+    results_newrows = []
+
+    # Getting today date
+    today = datetime.today()
+
+    # initializing the time count variable that Loops through the dates: today, yesterday, 1 month and six months
+    time_count = 0
+
+    # Initializing the Boolean has_timestamp that checks if the table has a timestamp
+    has_timestamp = False
+
+    # Initializing the more_info variable that will output extra information of the tables
+    more_info = ""
+
+    # Looping through all the table names for columns and rows match, and extra rows in Dev and prod
+    for table_name in table_names:
+        row_result = []
+        time_iteration = True
+        not_matched = False
+        not_matched_info = ""
+        newrows = [table_name]
+        while time_iteration:
+            # Checking if table has a timestamps and (if applicable) looping through today, yesterday, one month ago, and six month ago
+            if table_name in tables_with_timestamps:
+                if time_count == 0:
```

Comment:
I'm not sure I understand the `if time_count` logic... I think this would be more readable if you fed in a list of timestamps to use and looped through it. e.g. `DAYS_TO_CHECK = [one_day, one_month, six_months]` and then `for day_to_check in DAYS_TO_TEST: ` etc.